pkg/api/http_server.go

```diff
@@ -395,13 +403,18 @@ func (hs *HTTPServer) Run(ctx context.Context) error {
 		ReadTimeout: hs.Cfg.ReadTimeout,
 	}
 	switch hs.Cfg.Protocol {
-	case setting.HTTP2Scheme:
-		if err := hs.configureHttp2(); err != nil {
+	case setting.HTTP2Scheme, setting.HTTPSScheme:
+		if err := hs.configureTLS(); err != nil {
 			return err
 		}
-	case setting.HTTPSScheme:
-		if err := hs.configureHttps(); err != nil {
-			return err
+		if hs.Cfg.CertFile != "" && hs.Cfg.KeyFile != "" {
+			if hs.Cfg.CertWatchInterval != 0 {
```

Comment:
can `CertWatchInterval` be set to a negative duration? from what i can tell [time.ParseDuration](https://pkg.go.dev/time#ParseDuration) supports it

Reply:
i will address this comment. thanks

Reply:
One query, i have seen for all the duration parameters there are no checks being done about this.

Reply:
> no checks being done

i'm not sure i follow what you mean here, but the update in https://github.com/grafana/grafana/pull/83589/commits/942354659a2c3dfcac656892da1bc613a637e7b1 is what i was thinking of

Reply:
@toddtreece  what i meant, there are cfg parameters which are of type time.Duration. But i don't see any checks about the negative value in the existing code.